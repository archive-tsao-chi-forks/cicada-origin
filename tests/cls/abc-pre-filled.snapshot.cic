@def the_nat_type
@the (Trivial) -> Type
(_) => Nat

@class ABC [ a: Trivial = sole, b: the_nat_type(a), c: String ]

@def abc @the ABC(2, "c")
{ a: sole, b: 2, c: "c" }

@show abc

@show abc.a
@show abc.b
@show abc.c


@class ABCDE [ a: Trivial, b: the_nat_type(a), c: String = "c", d: String, e: String ]

@def abcde @the ABCDE(sole, 2, "d")
{ a: sole, b: 2, c: "c", d: "d", e: "e" }

@show abcde

@show abcde.a
@show abcde.b
@show abcde.c
@show abcde.d
@show abcde.e
