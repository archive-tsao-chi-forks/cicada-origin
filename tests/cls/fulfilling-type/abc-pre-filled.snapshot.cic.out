{
  a: sole
  b: 2
  c: "c"
}: class {
  a: Trivial = sole
  b: Nat = 2
  c: String
} 
sole: Trivial 
2: Nat 
"c": String 
{
  a: sole
  b: 2
  c: "d"
  d: "d"
  e: "e"
}: class {
  a: Trivial = sole
  b: Nat = 2
  c: String = "d"
  d: String
  e: String
} 
sole: Trivial 
2: Nat 
"d": String 
"d": String 
"e": String 

